### PR TITLE
Plane: added MAV_CMD_SET_HAGL support

### PR DESCRIPTION
**Pull request by tridge**
This allows for a lua script or external system over MAVLink to provide a height above ground level for landing of fixed wing or VTOL planes
This replaces #27262 as discussed on the dev call
requires this mavlink PR:
https://github.com/ArduPilot/mavlink/pull/362

### CODE REVIEW

**Comment by timtuxworth** [comment]
What would you use this for? I'm thinking of my terrain avoidance Lua work and wondering if it's relevant at all.

File: libraries/AP_Scripting/examples/land_hagl.lua [hunk 1]
```diff
@@ -0,0 +1,71 @@
+--[[
+   demonstrate proving HAGL to plane code for landing
+--]]
+
+---@diagnostic disable: param-type-mismatch
```
Comment (IamPete1) [inline libraries/AP_Scripting/examples/land_hagl.lua:5]: Cheating.

File: ArduPlane/GCS_Mavlink.cpp [hunk 2]
```diff
@@ -1083,6 +1083,10 @@ MAV_RESULT GCS_MAVLINK_Plane::handle_command_int_packet(const mavlink_command_in
         plane.set_mode(plane.mode_rtl, ModeReason::GCS_COMMAND);
         return MAV_RESULT_ACCEPTED;
 
+    case MAV_CMD_SET_HAGL:
```
Comment (peterbarker) [inline ArduPlane/GCS_Mavlink.cpp:1086]:
```suggestion
#if AP_MAVLINK_MAV_CMD_SET_HAGL_ENABLED
    case MAV_CMD_SET_HAGL:
```

File: libraries/AP_Scripting/examples/land_hagl.lua [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+   local accuracy = NaN()
+   gcs:run_command_int(MAV_CMD_SET_HAGL, { p1 = rangefinder_corrected, p2 = accuracy, p3=timeout_s })
+
+   if not last_active then
+      last_active = true
+      gcs:send_text(MAV_SEVERITY.INFO, string.format("HAGL Active %.1f", rangefinder_corrected))
+   end
+
+   -- log it
+   logger.write("HAGL", "RDist,HAGL", "ff", rangefinder_dist, rangefinder_corrected)
```
Comment (IamPete1) [inline libraries/AP_Scripting/examples/land_hagl.lua:60]:
```suggestion
   logger:write("HAGL", "RDist,HAGL", "ff", rangefinder_dist, rangefinder_corrected)
```

**Review by peterbarker — APPROVED** [review]
Good once the define is in place.

**Comment by Hwurzburg** [comment]
@tridge if we add a build_options.py dont forget the extract_features.py also

File: libraries/AP_Scripting/examples/land_hagl.lua (hunk 1 above)
Comment (tridge) [reply libraries/AP_Scripting/examples/land_hagl.lua]: fixed

File: ArduPlane/GCS_Mavlink.cpp (hunk 2 above)
Comment (tridge) [reply ArduPlane/GCS_Mavlink.cpp:1087]: done

File: libraries/AP_Scripting/examples/land_hagl.lua (hunk 3 above)
Comment (tridge) [reply libraries/AP_Scripting/examples/land_hagl.lua]: done

**Pull request merged**